### ISSUE TITLE
Using the API as a Standard User that is either a project member or owner cannot delete the namespace they have created

### ISSUE THREAD

**Issue by igomez06** [rancher/rancher#36717]
**Rancher Server Setup**
- Rancher version: v2.6-head commit: 635286172d41b24c074236060651af34c831b133
- Installation option (Docker install/Helm Chart): docker
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details: N/A

**Information about the Cluster**
- Kubernetes version:  v1.21.6+rke2r1
- Cluster Type (Local/Downstream): Local
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Standard User as either Project Owner or Project Member

   
**Describe the bug**

An admin user creates a project, and then, a standard user with the project role of  either project-owner or project-member creates a namespace, and then deletes that namespace they have created. The namespace creation is successful for both the owner and member, but fails for the deletion for both roles. With this error message
is forbidden: User "u-5nw27" cannot delete resource "namespaces" in API group "" in the namespace "testnamespace"

**To Reproduce**
 
1. v3/users POST to create a standard user
2. v3/projectroletemplatebindings POST to add either the project member or project owner role
3. v1/namespaces POST to create a namespace
4. v1/namespaces DELETE to delete a namespace

Or use the automation framework run test Also in the go automation framework run test `TestProjectUserTestSuite` found in `rancher/rancher/tests/v2/integration/project_user_test.go`
Make sure config file is properly setup please look at https://confluence.suse.com/display/RANQA/Rancher+Automation+Framework+for+Testing

**Result**
When attempting to delete the namespace I get this error `User "u-5nw27" cannot delete resource "namespaces" in API group "" in the namespace "testnamespace"`

**Expected Result**

To successfully delete the namespace that the project member or the project owner has created.

**Additional context**

This worked in 2.6.3

**Comment by MbolotSuse** [comment]
@igomez06 
I attempted to reproduce this through the UI and was unsuccessful. 

However, I was able to reproduce this using the automation framework. I think that the core of this issue is with the automation framework. If I place a breakpoint on the [namespace delete](https://github.com/rancher/rancher/blob/release/v2.6/tests/framework/extensions/namespaces/namespaces.go#L62), I'm able to get the tests to pass. I think that the core issue here is that the delete command (to remove the namespace) is being issued before the backend permissions (created by assigning the user to the project) have been created. There's a couple of ways that this could be solved (blocking return of the user assignment/namespace create until the permissions have been created, or putting a wait command in the namespace create). Either way, I think that the core issue lies with the automation framework rather than with rancher's RBAC.
